Meadowlark.net/Meadowlark.Net.Core/Middleware/ParsePathMiddleware.cs

```diff
@@ -0,0 +1,74 @@
+using System.Text.RegularExpressions;
+using Meadowlark.Net.Core.Model;
+
+namespace Meadowlark.Net.Core.Middleware;
+
+public static class ParsePathMiddleware
+{
+  private static string Decapitalize(string str)
+  {
+    if (str.Length == 0) return str;
+    if (str.Length == 1) return str.ToLower();
+    return str[0..1].ToLower() + str[1..];
+  }
+
+  private static PathComponents? PathComponentsFrom(string path)
+  {
+    // Matches all of the following sample expressions:
+    // /ed-fi/Sections
+    // /ed-fi/Sections/
+    // /ed-fi/Sections/idValue
+    Regex pathExpression = new(@"\/(?<projectNamespace>[^/]+)\/(?<endpointName>[^/]+)(\/|$)((?<documentUuid>[^/]*$))?");
```

Comment:
For full implementation, we'll want this regex to be compiled and defined in a lazy static property.